### PR TITLE
chore(typing): use types for DOM API

### PR DESCRIPTION
**Pull request by alexeagle**
This is possible now that ts2dart special cases these for dart.

Cannot merge this until https://github.com/angular/ts2dart/pull/232 is merged and released.

[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2804)

### CODE REVIEW

File: modules/angular2/src/dom/dom_adapter.ts [hunk 1]
```diff
@@ -47,16 +47,16 @@ export class DomAdapter {
   nodeValue(node): string { throw _abstract(); }
   type(node): string { throw _abstract(); }
   content(node): any { throw _abstract(); }
-  firstChild(el): /*(#2770) Node*/ any { throw _abstract(); }
-  nextSibling(el): /*(#2770) Node*/ any { throw _abstract(); }
-  parentElement(el): /*(#2770) Node*/ any { throw _abstract(); }
-  childNodes(el): List</*(#2770) Node*/ any> { throw _abstract(); }
-  childNodesAsList(el): List</*(#2770) Node*/ any> { throw _abstract(); }
+  firstChild(el): Node { throw _abstract(); }
```
Comment (mprobst) [inline modules/angular2/src/dom/dom_adapter.ts]: Also take `el: Element` and `node: Node`, `nodes: NodeList`, and so on?

**Comment by mprobst** [comment]
Nice, but needs more formatting ;-)

File: modules/angular2/src/dom/dom_adapter.ts (hunk 1 above)
Comment (alexeagle) [inline modules/angular2/src/dom/dom_adapter.ts]: Sure.
My larger scope is to crank up the typings one check at a time, so I was narrowing myself to return types first, and intend to come back for another pass to enforce parameter typings.
But for this PR may as well do both.

Comment (alexeagle) [inline modules/angular2/src/dom/dom_adapter.ts]: I tried this, and sadly it doesn't type-check. A bunch of methods here which take a parameter named 'el' actually need to accept the Node type. I'm not going to tackle that mess quite yet :)

**Comment by alexeagle** [comment]
merged as 27e7100

**Pull request closed without merging**